### ISSUE TITLE
Metals vscode Spark Project fails to start 

### ISSUE THREAD

**Issue by abhishekvaid** [scalameta/metals#1777]
**Describe the bug**
I have a simple spark project which isn't running in vscode. 

vscode version:
```
Version: 1.45.1
Commit: 5763d909d5f12fe19f215cbfdd29a91c0fa9208a
Date: 2020-05-14T08:33:47.663Z
Electron: 7.2.4
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Darwin x64 18.7.0
```
Scala version:
```
[spark-essentials]$ scala --version
Scala code runner version 2.13.1 -- Copyright 2002-2019, LAMP/EPFL and Lightbend, Inc.
[spark-essentials]$ 
```

metals version in vscode:
`1.9.0`

My build.sbt

```
name := "spark-essentials"

version := "0.1"

scalaVersion := "2.13.1"

val sparkVersion = "3.0.0-preview"
val vegasVersion = "0.3.11"
val postgresVersion = "42.2.2"

resolvers ++= Seq(
  "bintray-spark-packages" at "https://dl.bintray.com/spark-packages/maven",
  "Typesafe Simple Repository" at "https://repo.typesafe.com/typesafe/simple/maven-releases",
  "MavenRepository" at "https://mvnrepository.com"
)

libraryDependencies ++= Seq(
  "org.apache.spark" %% "spark-core" % sparkVersion,
  "org.apache.spark" %% "spark-sql" % sparkVersion,
  // logging
  "org.apache.logging.log4j" % "log4j-api" % "2.4.1",
  "org.apache.logging.log4j" % "log4j-core" % "2.4.1",
  // postgres for DB connectivity
  "org.postgresql" % "postgresql" % postgresVersion
)
```

Output for metals 
```

2020.05.24 17:23:30 INFO  running '/Library/Java/JavaVirtualMachines/jdk-11.0.4.jdk/Contents/Home/bin/java -Djline.terminal=jline.UnsupportedTerminal -Dsbt.log.noformat=true -Dfile.encoding=UTF-8 -jar /var/folders/f8/9f78kgk50jjd3d08h9hdqz_m0000gp/T/metals8346430227669979751/sbt-launch.jar -Dbloop.export-jar-classifiers=sources bloopInstall'
2020.05.24 17:23:41 INFO  [info] Loading global plugins from /Users/abhishekvaid/.sbt/1.0/plugins
2020.05.24 17:23:48 INFO  [info] Loading settings for project spark-essentials-build from metals.sbt ...
2020.05.24 17:23:48 INFO  [info] Loading project definition from /Users/abhishekvaid/GoogleDrive/moocs/udemy/daniel_ciocirlan/spark_rtjvm_01/spark-essentials/project
2020.05.24 17:23:53 INFO  [info] Loading settings for project spark-essentials from build.sbt ...
2020.05.24 17:23:53 INFO  [info] Set current project to spark-essentials (in build file:/Users/abhishekvaid/GoogleDrive/moocs/udemy/daniel_ciocirlan/spark_rtjvm_01/spark-essentials/)
2020.05.24 17:24:02 INFO  [warn] 
2020.05.24 17:24:02 INFO  [warn] 	Note: Unresolved dependencies path:
2020.05.24 17:24:02 INFO  [error] sbt.librarymanagement.ResolveException: Error downloading org.apache.spark:spark-sql_2.13:3.0.0-preview2
2020.05.24 17:24:02 INFO  [error]   Not found
2020.05.24 17:24:02 INFO  [error]   Not found
2020.05.24 17:24:02 INFO  [error]   not found: /Users/abhishekvaid/.ivy2/local/org.apache.spark/spark-sql_2.13/3.0.0-preview2/ivys/ivy.xml
2020.05.24 17:24:02 INFO  [error]   not found: https://repo1.maven.org/maven2/org/apache/spark/spark-sql_2.13/3.0.0-preview2/spark-sql_2.13-3.0.0-preview2.pom
2020.05.24 17:24:02 INFO  [error]   not found: https://dl.bintray.com/spark-packages/maven/org/apache/spark/spark-sql_2.13/3.0.0-preview2/spark-sql_2.13-3.0.0-preview2.pom
2020.05.24 17:24:02 INFO  [error]   not found: https://mvnrepository.com/org/apache/spark/spark-sql_2.13/3.0.0-preview2/spark-sql_2.13-3.0.0-preview2.pom
2020.05.24 17:24:02 INFO  [error]   not found: https://repo.typesafe.com/typesafe/simple/maven-releases/org/apache/spark/spark-sql_2.13/3.0.0-preview2/spark-sql_2.13-3.0.0-preview2.pom
2020.05.24 17:24:02 INFO  [error] Error downloading org.apache.spark:spark-core_2.13:3.0.0-preview2
2020.05.24 17:24:02 INFO  [error]   Not found
2020.05.24 17:24:02 INFO  [error]   Not found
2020.05.24 17:24:02 INFO  [error]   not found: /Users/abhishekvaid/.ivy2/local/org.apache.spark/spark-core_2.13/3.0.0-preview2/ivys/ivy.xml
2020.05.24 17:24:02 INFO  [error]   not found: https://repo1.maven.org/maven2/org/apache/spark/spark-core_2.13/3.0.0-preview2/spark-core_2.13-3.0.0-preview2.pom
2020.05.24 17:24:02 INFO  [error]   not found: https://dl.bintray.com/spark-packages/maven/org/apache/spark/spark-core_2.13/3.0.0-preview2/spark-core_2.13-3.0.0-preview2.pom
2020.05.24 17:24:02 INFO  [error]   not found: https://mvnrepository.com/org/apache/spark/spark-core_2.13/3.0.0-preview2/spark-core_2.13-3.0.0-preview2.pom
2020.05.24 17:24:02 INFO  [error]   not found: https://repo.typesafe.com/typesafe/simple/maven-releases/org/apache/spark/spark-core_2.13/3.0.0-preview2/spark-core_2.13-3.0.0-preview2.pom
2020.05.24 17:24:02 INFO  [error] 	at lmcoursier.CoursierDependencyResolution.unresolvedWarningOrThrow(CoursierDependencyResolution.scala:245)
2020.05.24 17:24:02 INFO  [error] 	at lmcoursier.CoursierDependencyResolution.$anonfun$update$34(CoursierDependencyResolution.scala:214)
2020.05.24 17:24:02 INFO  [error] 	at scala.util.Either$LeftProjection.map(Either.scala:573)
2020.05.24 17:24:02 INFO  [error] 	at lmcoursier.CoursierDependencyResolution.update(CoursierDependencyResolution.scala:214)
2020.05.24 17:24:02 INFO  [error] 	at sbt.librarymanagement.DependencyResolution.update(DependencyResolution.scala:60)
2020.05.24 17:24:02 INFO  [error] 	at sbt.internal.LibraryManagement$.resolve$1(LibraryManagement.scala:52)
2020.05.24 17:24:02 INFO  [error] 	at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$12(LibraryManagement.scala:102)
2020.05.24 17:24:02 INFO  [error] 	at sbt.util.Tracked$.$anonfun$lastOutput$1(Tracked.scala:69)
2020.05.24 17:24:02 INFO  [error] 	at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$20(LibraryManagement.scala:115)
2020.05.24 17:24:02 INFO  [error] 	at scala.util.control.Exception$Catch.apply(Exception.scala:228)
2020.05.24 17:24:02 INFO  [error] 	at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$11(LibraryManagement.scala:115)
2020.05.24 17:24:02 INFO  [error] 	at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$11$adapted(LibraryManagement.scala:96)
2020.05.24 17:24:02 INFO  [error] 	at sbt.util.Tracked$.$anonfun$inputChanged$1(Tracked.scala:150)
2020.05.24 17:24:02 INFO  [error] 	at sbt.internal.LibraryManagement$.cachedUpdate(LibraryManagement.scala:129)
2020.05.24 17:24:02 INFO  [error] 	at sbt.Classpaths$.$anonfun$updateTask0$5(Defaults.scala:2942)
2020.05.24 17:24:02 INFO  [error] 	at scala.Function1.$anonfun$compose$1(Function1.scala:49)
2020.05.24 17:24:02 INFO  [error] 	at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
2020.05.24 17:24:02 INFO  [error] 	at sbt.std.Transform$$anon$4.work(Transform.scala:67)
2020.05.24 17:24:02 INFO  [error] 	at sbt.Execute.$anonfun$submit$2(Execute.scala:280)
2020.05.24 17:24:02 INFO  [error] 	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:19)
2020.05.24 17:24:02 INFO  [error] 	at sbt.Execute.work(Execute.scala:289)
2020.05.24 17:24:02 INFO  [error] 	at sbt.Execute.$anonfun$submit$1(Execute.scala:280)
2020.05.24 17:24:02 INFO  [error] 	at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:178)
2020.05.24 17:24:02 INFO  [error] 	at sbt.CompletionService$$anon$2.call(CompletionService.scala:37)
2020.05.24 17:24:02 INFO  [error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2020.05.24 17:24:02 INFO  [error] 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
2020.05.24 17:24:02 INFO  [error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2020.05.24 17:24:02 INFO  [error] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
2020.05.24 17:24:02 INFO  [error] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
2020.05.24 17:24:02 INFO  [error] 	at java.base/java.lang.Thread.run(Thread.java:834)
2020.05.24 17:24:02 INFO  [error] (update) sbt.librarymanagement.ResolveException: Error downloading org.apache.spark:spark-sql_2.13:3.0.0-preview2
2020.05.24 17:24:02 INFO  [error]   Not found
2020.05.24 17:24:02 INFO  [error]   Not found
2020.05.24 17:24:02 INFO  [error]   not found: /Users/abhishekvaid/.ivy2/local/org.apache.spark/spark-sql_2.13/3.0.0-preview2/ivys/ivy.xml
2020.05.24 17:24:02 INFO  [error]   not found: https://repo1.maven.org/maven2/org/apache/spark/spark-sql_2.13/3.0.0-preview2/spark-sql_2.13-3.0.0-preview2.pom
2020.05.24 17:24:02 INFO  [error]   not found: https://dl.bintray.com/spark-packages/maven/org/apache/spark/spark-sql_2.13/3.0.0-preview2/spark-sql_2.13-3.0.0-preview2.pom
2020.05.24 17:24:02 INFO  [error]   not found: https://mvnrepository.com/org/apache/spark/spark-sql_2.13/3.0.0-preview2/spark-sql_2.13-3.0.0-preview2.pom
2020.05.24 17:24:02 INFO  [error]   not found: https://repo.typesafe.com/typesafe/simple/maven-releases/org/apache/spark/spark-sql_2.13/3.0.0-preview2/spark-sql_2.13-3.0.0-preview2.pom
2020.05.24 17:24:02 INFO  [error] Error downloading org.apache.spark:spark-core_2.13:3.0.0-preview2
2020.05.24 17:24:02 INFO  [error]   Not found
2020.05.24 17:24:02 INFO  [error]   Not found
2020.05.24 17:24:02 INFO  [error]   not found: /Users/abhishekvaid/.ivy2/local/org.apache.spark/spark-core_2.13/3.0.0-preview2/ivys/ivy.xml
2020.05.24 17:24:02 INFO  [error]   not found: https://repo1.maven.org/maven2/org/apache/spark/spark-core_2.13/3.0.0-preview2/spark-core_2.13-3.0.0-preview2.pom
2020.05.24 17:24:02 INFO  [error]   not found: https://dl.bintray.com/spark-packages/maven/org/apache/spark/spark-core_2.13/3.0.0-preview2/spark-core_2.13-3.0.0-preview2.pom
2020.05.24 17:24:02 INFO  [error]   not found: https://mvnrepository.com/org/apache/spark/spark-core_2.13/3.0.0-preview2/spark-core_2.13-3.0.0-preview2.pom
2020.05.24 17:24:02 INFO  [error]   not found: https://repo.typesafe.com/typesafe/simple/maven-releases/org/apache/spark/spark-core_2.13/3.0.0-preview2/spark-core_2.13-3.0.0-preview2.pom
2020.05.24 17:24:02 INFO  [error] Total time: 9 s, completed May 24, 2020, 5:24:02 PM
2020.05.24 17:24:02 INFO  build tool exit: 1
2020.05.24 17:24:03 INFO  time: ran 'sbt bloopInstall' in 32s
2020.05.24 17:24:03 ERROR sbt command failed: /Library/Java/JavaVirtualMachines/jdk-11.0.4.jdk/Contents/Home/bin/java -Djline.terminal=jline.UnsupportedTerminal -Dsbt.log.noformat=true -Dfile.encoding=UTF-8 -jar /var/folders/f8/9f78kgk50jjd3d08h9hdqz_m0000gp/T/metals8346430227669979751/sbt-launch.jar -Dbloop.export-jar-classifiers=sources bloopInstall
2020.05.24 17:24:03 ERROR Failed to connect with build server, no functionality will work.
java.lang.RuntimeException: The server did not start, got FailedToOpenBspConnection
	at bloop.launcher.LauncherMain.failPromise$1(Launcher.scala:95)
	at bloop.launcher.LauncherMain.runLauncher(Launcher.scala:122)
	at scala.meta.internal.metals.BloopServers$$anon$1.run(BloopServers.scala:107)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.lang.Thread.run(Thread.java:834)
```

**Comment by kpbochenek** [comment]
doesn't look like spark is released for scala 2.13 (https://mvnrepository.com/artifact/org.apache.spark/spark-core)

It doesn't work when you try build your `build.sbt` via sbt, it is not metals related.

**Comment by tgodzik** [comment]
Thanks for reporting! It does indeed seem like that version is not released, the sbt doesn't even build for me. We need a working sbt build to export all needed information that metals needs.
